turbinia/state_manager.py

```diff
@@ -132,14 +138,83 @@ def update_task(self, task):
       log.debug('Updating task {0:s} in Datastore'.format(task.name))
       self.client.put(entity)
 
-
   def write_new_task(self, task):
-    # Using the pubsub topic as part of the key in order to have unique entities
-    # per Turbinia installation.
     key = self.client.key('TurbiniaTask', task.id)
     entity = datastore.Entity(key)
     entity.update(self.get_task_dict(task))
     log.info('Writing new task {0:s} into Datastore'.format(task.name))
     self.client.put(entity)
     task.state_key = key
     return key
+
+
+class RedisStateManager(BaseStateManager):
+  """Use redis for task state storage.
+
+  Attributes:
+    client: Redis database object.
+  """
+
+  def __init__(self):
+    config.LoadConfig()
+    self.client = redis.StrictRedis(
+        host=config.REDIS_HOST,
+        port=config.REDIS_PORT,
+        db=config.REDIS_DB)
+
+  def get_task_data(self, instance, days=0, task_id=None, request_id=None):
+    """Gets task data from Redis.
+
+    Args:
+      instance (string): The Turbinia instance name (by default the same as the
+          PUBSUB_TOPIC in the config).
+      days (int): The number of days we want history for.
+      task_id (string): The Id of the task.
+      request_id (string): The Id of the request we want tasks for.
+
+    Returns:
+      List of Task dict objects.
+    """
+    tasks = [json.loads(self.client.get(task))
+             for task in self.client.scan_iter('TurbiniaTask:*')
+             if json.loads(self.client.get(task)).get('instance') == instance
+             or not instance]
+    if days:
+      start_time = datetime.now() - timedelta(days=days)
+      return [task for task in tasks
+              if datetime.strptime(task.get('last_update'), '%Y-%m-%dT%H:%M:%S')
+              > start_time]
+    elif task_id:
+      return [task for task in tasks if task.get('task_id') == task_id]
+    elif request_id:
+      return [task for task in tasks if task.get('request_id') == request_id]
+    return tasks
+
+  def update_task(self, task):
+    key = task.state_key
+    if not self.client.get(key):
+      self.write_new_task(task)
+      return
+    log.info('Updating task {0:s} in Datastore'.format(task.name))
+    task_data = self.get_task_dict(task)
+    task_data['last_update'] = task_data['last_update'].strftime(
```

Comment:
Is this getting converted back and forth between datetime and string because of the json serialization?  Or does redis just not support datetime?  Can you maybe add a comment about this in either case?